### PR TITLE
node: terra2 support

### PR DESCRIPTION
**Pull request by evan-gray**
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1282)

### CODE REVIEW

File: node/cmd/guardiand/node.go [hunk 1]
```diff
@@ -534,6 +543,15 @@ func runNode(cmd *cobra.Command, args []string) {
...
 	if *testnetMode || *unsafeDevMode {
+		if *terra2WS == "" {
+			logger.Fatal("Please specify --terra2WS")
+		}
+		if *terra2LCD == "" {
+			logger.Fatal("Please specify --terra2LCD")
+		}
+		if *terra2Contract == "" {
+			logger.Fatal("Please specify --terra2Contract")
+		}
```
Comment (bruce-riley) [inline node/cmd/guardiand/node.go:554]: This says we must run terra2 in testnet, so as soon as this gets merged the next testnet deploy must have terra2. Is that what we want?

File: node/pkg/terra/watcher.go [hunk 2]
```diff
@@ -115,21 +136,21 @@ func (e *Watcher) Run(ctx context.Context) error {
...
 
 	// Wait for the success response
 	_, _, err = c.ReadMessage()
 	if err != nil {
-		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDTerra, 1)
+		p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
 		terraConnectionErrors.WithLabelValues("event_subscription_error").Inc()
 		return fmt.Errorf("event subscription failed: %w", err)
 	}
 	logger.Info("subscribed to new transaction events")
```
Comment (bruce-riley) [inline node/pkg/terra/watcher.go:151]: In the Eth watcher, we have a networkName which is included in all the log messages so we can tell which chain the error is coming from. Do we want to start doing that for Terra now?

**Review by bruce-riley — APPROVED** [review]

File: node/cmd/guardiand/node.go (hunk 1 above)
Comment (SEJeff) [reply node/cmd/guardiand/node.go:554]: Yes I think so. The terra 2 node was setup today and should be done being configured tomorrow in theory.

File: node/pkg/terra/watcher.go [hunk 3]
```diff
@@ -83,12 +92,24 @@ func NewWatcher(
...
+	chainID vaa.ChainID) *Watcher {
+
+	// CosmWasm 1.0.0
+	contractAddressFilterKey := "execute._contract_address"
+	contractAddressLogKey := "_contract_address"
+	if chainID == vaa.ChainIDTerra {
+		// CosmWasm <1.0.0
+		contractAddressFilterKey = "execute_contract.contract_address"
+		contractAddressLogKey = "contract_address"
+	}
```
Comment (SEJeff) [inline node/pkg/terra/watcher.go:106]: These bits are utter magic to me.

no idea dog

File: node/pkg/terra/watcher.go (hunk 2 above)
Comment (SEJeff) [reply node/pkg/terra/watcher.go:151]: Yes, yes we absolutely do!

**Review by SEJeff — APPROVED** [review]
I'd like to see you take a stab and the node/hack/repair_terra port to terra2.

GIven how much fun we have, I'm strongly hesitant to add chains that we can't easily repair.

File: node/pkg/terra/watcher.go (hunk 3 above)
Comment (evan-gray) [reply node/pkg/terra/watcher.go:106]: lol that's what was in the watcher before (for terra classic) but CosmWasm 1.0.0 has different fields for the contract address on logs and filters https://github.com/terra-money/core/blob/main/WASM_MIGRATE.md#execute

File: node/pkg/terra/watcher.go (hunk 2 above)
Comment (evan-gray) [reply node/pkg/terra/watcher.go:153]: thanks! added `terra_network` to the metric (possibly to be renamed in the future)

**Comment by evan-gray** [comment]
Note: the following prometheus metrics now include `terra_network`
- `wormhole_terra_connection_errors_total`
- `wormhole_terra_messages_confirmed_total`
- `wormhole_terra_current_height`
- `wormhole_terra_query_latency`
This also changes the middle two to `Vec`s.

**Review by SEJeff — APPROVED** [review]

**Review by SEJeff — APPROVED** [review]

**Review by bruce-riley — APPROVED** [review]

**Pull request merged**